custom_commands.py

```diff
@@ -83,58 +87,94 @@ def when_not_matched_then_insert(self):
+    """
+    def __init__(self, file_names: List[str]) -> None:
+        self.file_names = file_names
+
+    def __str__(self) -> str:
+        the_files = ["'" + f.replace("'", "\\'") + "'" for f in self.file_names]
+        return f"({','.join(the_files)})"
+
+
 class CopyInto(UpdateBase):
     """Copy Into Command base class, for documentation see:
     https://docs.snowflake.net/manuals/sql-reference/sql/copy-into-location.html"""
 
     __visit_name__ = 'copy_into'
     _bind = None
 
-    def __init__(self, from_, into, formatter):
+    def __init__(self, from_, into, formatter=None):
         self.from_ = from_
         self.into = into
         self.formatter = formatter
         self.copy_options = {}
 
-    def __repr__(self):
-        options = (' ' + ' '.join(["{} = {}".format(n, str(v)) for n, v in
-                                   self.copy_options.items()])) if self.copy_options else ''
-        return "COPY INTO {} FROM {} {}{}".format(self.into.__repr__(),
-                                                  self.from_.__repr__(),
-                                                  self.formatter.__repr__(),
-                                                  options)
-
     def bind(self):
         return None
 
-    def overwrite(self, overwrite):
-        if not isinstance(overwrite, bool):
-            raise TypeError("Parameter overwrite should  be a boolean value")
-        self.copy_options.update({'OVERWRITE': translate_bool(overwrite)})
+    def force(self, force):
+        if not isinstance(force, bool):
+            raise TypeError("Parameter force should  be a boolean value")
+        self.copy_options.update({'FORCE': translate_bool(force)})
+        return self
 
     def single(self, single_file):
         if not isinstance(single_file, bool):
             raise TypeError("Parameter single_file should  be a boolean value")
         self.copy_options.update({'SINGLE': translate_bool(single_file)})
+        return self
 
     def maxfilesize(self, max_size):
         if not isinstance(max_size, int):
             raise TypeError("Parameter max_size should be an integer value")
         self.copy_options.update({'MAX_FILE_SIZE': max_size})
+        return self
+
+    def files(self, file_names: List[str]) -> 'CopyInto':
+        self.copy_options.update({'FILES': FilesOption(file_names)})
+        return self
+
+    def pattern(self, pattern: str) -> 'CopyInto':
+        self.copy_options.update({'PATTERN': pattern})
+        return self
 
 
 class CopyFormatter(ClauseElement):
+    """
+    Base class for Formatter specifications inside a COPY INTO statement. May also
+    be used to create a named format.
+    """
     __visit_name__ = 'copy_formatter'
 
-    def __init__(self):
-        self.options = {}
+    def __init__(self, format_name=None):
+        if format_name:
+            self.options = {"format_name": format_name}
+        else:
+            self.options = {}
 
-    def __repr__(self):
```

Comment:
> > Made modifications as described in the comments; please re-review.
> 
> I'm sorry @martin-haffner-by but it seems like most of my comments were not resolved is that on purpose? I resolved the only type-hint comments, but there are still others.
> 
> Also I have pulled in your changes into our monorepo where source code still mainly lives and I am running tests right now.

You're right, sorry, I missed some of your comments. Gonna update this right now.

Reply:
> Can we update the `__repr__` function instead of removing it?

OK. I have changed the `__repr__` functions so that they output the important information, and added a comment that this is meant for logging / debugging purposes and not for compilation, so that no one feels tempted to use just `repr(command)` instead of `command.compile(dialect)`.

````
        """
        repr for debugging / logging purposes only. For compilation logic, see
        the corresponding visitor in base.py
        """
        return f"COPY INTO {self.into} FROM {self.from_} {self.formatter} {self.options}"
````
OK?